### ISSUE TITLE
Submit business contact information - full version

### ISSUE THREAD

**Issue by findingsimple** [woocommerce/google-listings-and-ads#962]
From August 9th the Merchant Center required additional contact information to be submitted during MC setup. 

[Previous issue here](https://github.com/woocommerce/google-listings-and-ads/issues/863)

We implemented the lite version of the flow while A[PIs for phone verification](https://developers.google.com/shopping-content/guides/phoneverification) are being implemented.

[The Phone verification API is now available](https://developers.google.com/shopping-content/guides/phoneverification)

We can now adjust the MC onboarding flow to account for the full version.

Link to internal design explorations

pbIy4N-1JA-p2

Link to Figma file below.

We don't have a fixed date for the policy enforcement yet on the Google side - will likely be October.

## Technical

[Phone Verification API](https://developers.google.com/shopping-content/guides/phoneverification)

### Frontend development tasks

- New component
    - [x] Implement a new component for inputting verification code
- `<PhoneNumberCard>` component
    - [ ] ~Add verified state and its UI~ (could not implement it before getting the verification state)
    - [x] Add verification process of text message
    - [x] Add verification process of phone call
    - [x] Integrate with verification API and handle error responses
    - [x] Propagate the verified states of phone number to outer scope via prop callback
- `<StoreAddressCard>` component
    - [x] Adjust the layout to fit with the new visual design
- `<StoreRequirements>` component
    - [x] Remove the API calling of saving phone number. (It will be handled within the `<PhoneNumberCard>`)
    - [x] Add the verified states of phone number to the disabled control of submit button
- `<EditContactInformation>` component
    - [x] Remove the API calling of saving phone number. (It will be handled within the `<PhoneNumberCard>`)
    - [x] Split the editing of phone number and store address into individual sub-pages
- `<ContactInformationPreview>` component
    - [ ] ~Distinguish verified and unverified states~ (could not implement it before getting the verification state)
    - [x] Implement new `view` to split the phone number card and store address card, which is used in the Settings page https://github.com/woocommerce/google-listings-and-ads/pull/1025
- Clean up
    - [x] Clean up all TODO comments by searching `TODO: [full-contact-info]`

### Backend development tasks

- [x] Add phone verification service (https://github.com/woocommerce/google-listings-and-ads/issues/980)
- [x] Refactor ContactInformation Service and API (https://github.com/woocommerce/google-listings-and-ads/issues/981)
- [x] Add API endpoints for requesting phone verification code and verifying it (https://github.com/woocommerce/google-listings-and-ads/issues/982)

## Figma link

29d12-pb/

## Event tracking

- TBC

**Comment by eason9487** [comment]
Hi @j111q, (please let me know if I should ping Shiva 🙏 )

I think the error cases of verification codes might happen more frequently than other operations, such as incorrect codes or expired. Do we still show these errors to users via the error notification popup at the left-bottom side of page?

**Comment by eason9487** [comment]
Note for backend devs:

Currently, the phone number and store address are updated via the same API at the same time. But there is a separate process to verify phone number and get back the verification result in the full version. The design or change of this API might have a more significant impact scope. We probably need to clarify it further and earlier. Please let me know if it needs more current/future implementation details of the frontend side.

**Comment by j111q** [comment]
@eason9487 yes do ping me only, that's fine! 😄 

Hmmmm yeah I was just thinking about it too. At first I was assuming we would just use the little pop-up, but then I realised it would probably be better in context, since it's not an error for the entire page 🤔 

How about something like this: 

![image](https://user-images.githubusercontent.com/65582817/129878650-04c55f38-950a-47f7-82e2-f8303edfff75.png)

(If we can be more descriptive with errors it would be good, will leave you to chime in on the exact copy.)

**Comment by eason9487** [comment]
@j111q, thanks for the reply!

How about show the error at this UI state? Therefore, users would no need to redo the verification method selection if they keyed in wrong verification codes. Or click the "Resend code" button if it has expired.

![image](https://user-images.githubusercontent.com/17420811/129879472-5b0c3b48-f490-4454-89e9-72c0bb60e482.png)

And at the same UI state for the phone call method.

**Comment by j111q** [comment]
Hmm yes possible. 

Was just thinking we could use one error state for both verification methods for simplicity. 
Also, this might be easier if the user needs to **switch** verification methods. In the current design, if you want to switch methods, it's not clear how to go back... You might have to refresh the page, which people might not want to do out of fear they lose all their progress. Thoughts?

**Comment by eason9487** [comment]
> In the current design, if you want to switch methods, it's not clear how to go back... You might have to refresh the page, which people might not want to do out of fear they lose all their progress. Thoughts?

Would it work if we change the text of the "Edit" button to something like "Back" or "Change phone number or verification method"?

![image](https://user-images.githubusercontent.com/17420811/129879472-5b0c3b48-f490-4454-89e9-72c0bb60e482.png)

**Comment by j111q** [comment]
Hmmmmmm. Okay I tried out a few iterations here and all seemed a little weird.
This one felt like it works best to me:

![image](https://user-images.githubusercontent.com/65582817/130029010-0f87b02b-883b-49c3-9657-9b9f83ee5238.png)

It would be a new "footer" to the card, that shows all the time, even without any error:

![image](https://user-images.githubusercontent.com/65582817/130029081-04b47d33-c88b-44e6-8db1-36bdd5912476.png)

What do you think?

**Comment by eason9487** [comment]
Thank you @j111q. It looks good to me, and it has consistent UI and behavior with the MC account connection. 👍 

![image](https://user-images.githubusercontent.com/17420811/130031229-160d0557-b7d3-4121-a1e9-b0f121d050f7.png)

**Comment by j111q** [comment]
Yes haha okay cool 👍  Thanks Eason!

**Comment by nima-karimi** [comment]
Should we allow users to save an unverified phone number or do we only save a phone number after verification?

Merchant Center seems to allow users to save a phone number and displays it as unverified. Would this policy change after rollout?

cc: @j111q

**Comment by nima-karimi** [comment]
Added the backend tasks required for the phone verification functionality to the PR description (#980, #981, #982).

**Comment by eason9487** [comment]
Hey @j111q,

Since the data returned by Google API for both verified and unverified phone numbers are the same, we still cannot distinguish whether it is verified or not. All the logic associated with the verification status would not be able to implement. We might need to make some adjustments. For examples:
- No matter the user has saved/verified their phone number at Goolge Merchant Center, they would always be asked to verify phone number at GLA plugin when onboarding.
- The verified and unverified status of phone number might not be able to display on UI.
- All prompts in inbox notification, issue list, and settings page are only available when the phone number doesn't exist.

cc @jconroy 

---

Questions about UI:

1. The phone number would be saved once it gets verified via Goolge API, so the "Save changes" button on the Setting page will only interact with the store address data. I was thinking maybe we could change the wording of that button, or Make the save button and the store address card appear as a set on the display to imply to users that the button is only acting on the address data.
2. Could we keep displaying the mobile icon after the phone being verified on the Settings page? Thus it could have the consistency of the icon display during all edit phone number operations. The last diagram of edit phone number on the Setting page:
    ![image](https://user-images.githubusercontent.com/17420811/131781545-ece4622b-e09a-49be-9140-a6504a9a63fb.png)
    (it's not the contact information card on the main Settings page)

**Comment by j111q** [comment]
> Since the data returned by Google API for both verified and unverified phone numbers are the same, we still cannot distinguish whether it is verified or not. 

I was under the impression that Google would definitely provide this - @jconroy can we maybe confirm with Peter? 

Without this data, we basically can't tell which user has a verified number and which user doesn't have one, right? If that's the case, all we can do is prompt ALL users to verify, and then .... cross our fingers and hope they verified it. We have no way to nudge those with unverified number to verify - or we risk prompting people who already have verified numbers to verify it again and again.

**Comment by j111q** [comment]
> Could we keep displaying the mobile icon after the phone being verified on the Settings page? Thus it could have the consistency of the icon display during all edit phone number operations. The last diagram of edit phone number on the Setting page:

Yes, I think we should too 😅  Good catch

**Comment by eason9487** [comment]
@j111q 

> Without this data, we basically can't tell which user has a verified number and which user doesn't have one, right?

Right. So it'd be great if Google could provide that information.

**Comment by j111q** [comment]
Okay got it. I added it to our weekly call agenda with Google to just clarify and see whether there's any solution here. Will circle back on Wednesday after we talk to them! Thanks @eason9487  🙏

**Comment by findingsimple** [comment]
> Since the data returned by Google API for both verified and unverified phone numbers are the same, we still cannot distinguish whether it is verified or not.

@caiotramontina @Kheiden hey guys - wondering if you would know what is the recommended way to tell if a user has verified?

**Comment by j111q** [comment]
Comment in another Google doc from @caiotramontina:

> Apparently there is no direct method to retrieve the status (waiting for a final confirmation). However, in case phone number has not been verified, this will become **an account-level issue of type PENDING_PHONE_VERIFICATION**. One alternative would be use that issue to indicate that phone number has not been verified (with that approach, the extension will be able to know when it's not verified, but will not be able to know if it is verified).

Given the above ☝️ , I have tweaked the Figma file and removed the places where it shows `Verified`, and instead we will only show `Unverified`. As Caio mentioned, `Unverified` should appear when there is an account level issue of type: `PENDING_PHONE_VERIFICATION`. I think we shouldn't show `Verified` if we don't definitively know that it IS verified. 

LMK if this works @eason9487 🙏

**Comment by eason9487** [comment]
Thanks for replying, @j111q

> Apparently there is no direct method to retrieve the status (waiting for a final confirmation). However, in case phone number has not been verified, this will become **an account-level issue of type PENDING_PHONE_VERIFICATION**. One alternative would be use that issue to indicate that phone number has not been verified

Not sure where I did wrong. I tried two different Google Merchant Center accounts (359541474 and 456021378) by [this doc](https://developers.google.com/shopping-content/reference/rest/v2.1/accountstatuses/get), but no matter the phone numbers were empty or unverified, none of them got the account level issue with `PENDING_PHONE_VERIFICATION` type. It doesn't appear in the Google Merchant Center either:

![image](https://user-images.githubusercontent.com/17420811/133047312-27a0636b-0eba-485e-85e4-8ea31eb80504.png)

**Comment by j111q** [comment]
Hmmm noted @eason9487. Pinged Caio and CCed you.

**Comment by caiotramontina** [comment]
This is probably due to the fact that enforcement has not yet started. I've asked for internal confirmation that this is in fact the correct enum that will be utilized as id of the AccountLevelIssue and if there is anyway to create a test environment in the meantime. Will followup as soon I have an answer.

> Not sure where I did wrong. I tried two different Google Merchant Center accounts (359541474 and 456021378) by [this doc](https://developers.google.com/shopping-content/reference/rest/v2.1/accountstatuses/get), but no matter the phone numbers were empty or unverified, none of them got the account level issue with `PENDING_PHONE_VERIFICATION` type. It doesn't appear in the Google Merchant Center either:

**Comment by j111q** [comment]
FYI (to record this here) - I have split the design to “Before verification can be detected” and “After verification can be detected” - let me know if it makes sense. https://www.figma.com/file/eQ9O4m2flzrcAiDMXkOW0m/?node-id=1%3A145

**Comment by eason9487** [comment]
After discussing with @j111q about this question quoted from https://github.com/woocommerce/google-listings-and-ads/issues/962#issuecomment-911157548

> Questions about UI:
> 
> 1. The phone number would be saved once it gets verified via Goolge API, so the "Save changes" button on the Setting page will only interact with the store address data. I was thinking maybe we could change the wording of that button, or Make the save button and the store address card appear as a set on the display to imply to users that the button is only acting on the address data.

The changes are

- adjust the layout of `<StoreAddressCard>` for both usages in MC onboarding and GLA Settings pages
- split the editing of phone number and store address into individual sub-pages in GLA Settings

**Comment by tomalec** [comment]
@j111q I guess we need to separate missing data warning as well 2a4c0-pb/#bash
![image](https://user-images.githubusercontent.com/17435/134676102-0f5a496e-a0c0-4cd1-8697-dc30a6f5ebe4.png)
As "Add information" would have to point to two separate pages at once.

**Comment by j111q** [comment]
Oh great catch, thanks for that @tomalec. 
Can we do it like this? Updated the design in Figma as well.

![image](https://user-images.githubusercontent.com/65582817/134708622-9a5ea7f8-6766-4c56-a0c0-58201b858591.png)

**Comment by tomalec** [comment]
Thanks, that's more or less what I already started implementing.

WDYT of moving warning icon to the same line? to make the height of the card closer to the one w/o the message, to make the layout less jumpy as it loads.
![image](https://user-images.githubusercontent.com/17435/134718643-5f801aea-d5e4-40fe-9fac-58f800284877.png)


There is still a difference, but that's only one line of text more adds 18px to the height. "Phone number" and "(!) Please add your store address" headers ar the same height.

**Comment by tomalec** [comment]
Also, I experimented a little with preloaders

#### Previous:

![spinner](https://user-images.githubusercontent.com/17435/134724207-cbd14797-8077-43cf-89ba-014c08a5a947.gif)
#### Proposed:
![inline preloader](https://user-images.githubusercontent.com/17435/134723533-1f239abd-77fe-4d12-8dff-ab369ebdc9bd.gif)
IMHO, it gives the user the size and shape of data to anticipate, and is less mesmerizing.
Also, it reduces the layout shift.

There still be one in case of a warning, but at least we reduce it for the happy-path 
![inline warning](https://user-images.githubusercontent.com/17435/134723695-867b459c-ed5f-48c2-9aaf-45c8b8b012cc.gif)


WDYT?

**Comment by tomalec** [comment]
@j111q I updated the Store Address setting card:
The one in Figma looks like:
![image](https://user-images.githubusercontent.com/17435/135344395-cfa1234f-dc60-4bae-a8a5-eed6cb852e3f.png)

So far we are using an external icon set for the "reload" icon:
![image](https://user-images.githubusercontent.com/17435/135344474-c43135ff-5a87-4241-877e-b601484f430a.png)


I noticed we have "Update" icon in the [Gutenberg set](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library).
![image](https://user-images.githubusercontent.com/17435/135344901-5f84d880-1e71-413c-9134-dea8344ed202.png)

WDYT of using it?
It could save few bytes of code delivered to the merchant's browser and hopefully give a more consistent experience.
